pycbc/transforms.py

```diff
@@ -2501,3 +2501,61 @@ def read_transforms_from_config(cp, section="transforms"):
         t = transforms[name].from_config(cp, section, subsection)
         trans.append(t)
     return order_transforms(trans)
+
+
+def new_z_to_euler(new_z):
+    """
+    From the new Z axis expressed in (polar, azimuthal) angles of the
+    initial coordinate system, return the (alpha, beta) Euler angles
+    that rotate the old Z axis (0, 0) to the new Z axis.
+    """
+    return (0.5*numpy.pi + new_z[1]) % (2 * numpy.pi), new_z[0]
+
+
+def rotate_euler(sph_coords, alpha, beta, gamma):
+    """
+    Take an Nx2 array of N (theta, phi) vectors on the unit sphere
+    (that is, (polar, azimuthal) angles in radians) and apply
+    rotations through the Euler angles alpha, beta, and gamma
+    in radians, using the ZXZ convention.
+    """
+    c = numpy.cos
+    s = numpy.sin
+
+    # Define rotation matrix
+    R = numpy.array(
+        [[c(alpha) * c(gamma) - c(beta) * s(alpha) * s(gamma),
+          c(gamma) * s(alpha) + c(alpha) * c(beta) * s(gamma),
+          s(beta) * s(gamma)],
+         [-c(beta) * c(gamma) * s(alpha) - c(alpha) * s(gamma),
+          c(alpha) * c(beta) * c(gamma) - s(alpha) * s(gamma),
+          c(gamma) * s(beta)],
+         [s(alpha) * s(beta), -c(alpha) * s(beta), c(beta)]], dtype=float)
+
+    # Convert to intermediate cartesian representation (N x 3)
+    cart_orig = numpy.empty(shape=(len(sph_coords), 3), dtype=float)
+    cart_orig[:, 0] = c(sph_coords[:, 1])
+    cart_orig[:, 1] = s(sph_coords[:, 1])
+    cart_orig[:, 0:2] *= s(sph_coords[:, 0])[:, None]
+    cart_orig[:, 2] = c(sph_coords[:, 0])
+
+    # Rotate by x_i = R_{ij} * x_j
+    # NB: dot contracts last dim of A with second-to-last dim of B
+    cart_new = numpy.dot(cart_orig, R)
+
+    # Extract new spherical coordinates
+    sph_new = numpy.empty(shape=(len(sph_coords), 2), dtype=float)
+    sph_new[:, 0] = numpy.arccos(cart_new[:, 2])
+    sph_new[:, 1] = numpy.arctan2(cart_new[:, 1], cart_new[:, 0])
+
+    return sph_new
+
+
+def decra2polaz(dec, ra):
```

Comment:
Would this make more sense in the conversions module and with naming to match the style there?

Reply:
That seems appropriate indeed.

Reply:
@Prasia-Pankunni could you move the `decra2polaz` and `polaz2radec` functions to `pycbc/conversions.py` instead? I would put them at line 859, inside the "Extrinsic parameter functions" section. Remember to update the imports accordingly.